modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java

```diff
@@ -138,42 +138,54 @@ public void processOpts() {
         if (additionalProperties.containsKey(WITH_GO_CODEGEN_COMMENT)) {
             setWithGoCodegenComment(Boolean.parseBoolean(additionalProperties.get(WITH_GO_CODEGEN_COMMENT).toString()));
             if (withGoCodegenComment) {
-                additionalProperties.put(WITH_GO_CODEGEN_COMMENT, "true");
+                additionalProperties.put(WITH_GO_CODEGEN_COMMENT, true);
+            } else {
+                additionalProperties.put(WITH_GO_CODEGEN_COMMENT, false);
```

Comment:
why not simply `additionalProperties.put(WITH_GO_CODEGEN_COMMENT, withGoCodegenComment);`? same for other rows

Reply:
Good point. I will make the change.